namespace/namespace.go

```diff
@@ -6,55 +6,60 @@ import (
 	"github.com/sourcenetwork/corekv"
 )
 
-// namespaceStore wraps a namespace of another database as a logical database.
-type namespaceStore struct {
+// Datastore wraps a namespace of another database as a logical database.
+type Datastore struct {
```

Comment:
question: Do we really need a non-transaction namespace store? Both badger and memory support transactions so namespace could be removed and this would reduce the usage and maintenance overhead.

Reply:
That would remove the ability to namespace stores that dont support transactions.  Including stores defined outside of the corekv repo.

I think that is quite a hit given the very low cost of supporting that flexibility (note: corekv is not a Defra-only project).